[CI] DataLifecycleServiceIT testErrorRecordingOnRollover failing

**Build scan:**
https://gradle-enterprise.elastic.co/s/njctqv4g2k7ly/tests/:modules:dlm:internalClusterTest/org.elasticsearch.dlm.DataLifecycleServiceIT/testErrorRecordingOnRollover

**Reproduction line:**
```
gradlew ':modules:dlm:internalClusterTest' --tests "org.elasticsearch.dlm.DataLifecycleServiceIT.testErrorRecordingOnRollover" -Dtests.seed=7A0C4AB39ECCF42F -Dtests.locale=fr-FR -Dtests.timezone=Universal -Druntime.java=20
```

**Applicable branches:**
main

**Reproduces locally?:**
Didn't try

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.dlm.DataLifecycleServiceIT&tests.test=testErrorRecordingOnRollover

**Failure excerpt:**
```
java.lang.AssertionError: 
Expected: null
     but: was "{\"type\":\"exception\",\"reason\":\"Force merge request only had 1 successful shards out of a total of 2\"}"

  at __randomizedtesting.SeedInfo.seed([7A0C4AB39ECCF42F:EF340B4E82C2DA79]:0)
  at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
  at org.junit.Assert.assertThat(Assert.java:956)
  at org.junit.Assert.assertThat(Assert.java:923)
  at org.elasticsearch.dlm.DataLifecycleServiceIT.lambda$testErrorRecordingOnRollover$11(DataLifecycleServiceIT.java:453)
  at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:1158)
  at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:1131)
  at org.elasticsearch.dlm.DataLifecycleServiceIT.testErrorRecordingOnRollover(DataLifecycleServiceIT.java:435)
  at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
  at java.lang.reflect.Method.invoke(Method.java:578)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:1623)

```

Another one at https://gradle-enterprise.elastic.co/s/yce2eulrkrahu/tests/:modules:dlm:internalClusterTest/org.elasticsearch.dlm.DataLifecycleServiceIT/testErrorRecordingOnRollover?top-execution=1

```
java.lang.AssertionError: |  
-- | --
  | Expected: null |  
  | but: was "{\"type\":\"exception\",\"reason\":\"Force merge request only had 1 successful shards out of a total of 2\"}" |  

...
REPRODUCE WITH: ./gradlew ':modules:dlm:internalClusterTest' --tests "org.elasticsearch.dlm.DataLifecycleServiceIT.testErrorRecordingOnRollover" -Dtests.seed=258BA403A0434690 -Dtests.locale=ar-LB -Dtests.timezone=Asia/Makassar -Druntime.java=20
```

Will mute.

--- a/modules/dlm/src/internalClusterTest/java/org/elasticsearch/dlm/DataLifecycleServiceIT.java
+++ b/modules/dlm/src/internalClusterTest/java/org/elasticsearch/dlm/DataLifecycleServiceIT.java
@@ -541,7 +541,16 @@ public class DataLifecycleServiceIT extends ESIntegTestCase {
                 // error stores don't contain anything for the first generation index anymore
                 Iterable<DataLifecycleService> lifecycleServices = internalCluster().getInstances(DataLifecycleService.class);
                 for (DataLifecycleService lifecycleService : lifecycleServices) {
-                    assertThat(lifecycleService.getErrorStore().getError(firstGenerationIndex), nullValue());
+                    if (internalCluster().numDataNodes() > 1) {
+                        assertThat(lifecycleService.getErrorStore().getError(firstGenerationIndex), nullValue());
+                    } else {
+                        if (lifecycleService.getErrorStore().getError(firstGenerationIndex) != null) {
+                            assertThat(
+                                lifecycleService.getErrorStore().getError(firstGenerationIndex),
+                                containsString("Force merge request only had 1 successful shards out of a total of 2")
+                            );
+                        }
+                    }
                 }
             });
         } finally {

--- a/modules/dlm/src/internalClusterTest/java/org/elasticsearch/dlm/DataLifecycleServiceIT.java
+++ b/modules/dlm/src/internalClusterTest/java/org/elasticsearch/dlm/DataLifecycleServiceIT.java
@@ -30,6 +30,7 @@ import org.elasticsearch.cluster.metadata.ComposableIndexTemplate;
 import org.elasticsearch.cluster.metadata.DataLifecycle;
 import org.elasticsearch.cluster.metadata.DataStream;
 import org.elasticsearch.cluster.metadata.DataStreamAction;
+import org.elasticsearch.cluster.metadata.IndexMetadata;
 import org.elasticsearch.cluster.metadata.Template;
 import org.elasticsearch.cluster.service.ClusterService;
 import org.elasticsearch.common.compress.CompressedXContent;
@@ -385,7 +386,14 @@ public class DataLifecycleServiceIT extends ESIntegTestCase {
         // empty lifecycle contains the default rollover
         DataLifecycle lifecycle = new DataLifecycle();
 
-        putComposableIndexTemplate("id1", null, List.of("metrics-foo*"), null, null, lifecycle);
+        putComposableIndexTemplate(
+            "id1",
+            null,
+            List.of("metrics-foo*"),
+            Settings.builder().put(IndexMetadata.SETTING_AUTO_EXPAND_REPLICAS, "0-1").build(),
+            null,
+            lifecycle
+        );
         Iterable<DataLifecycleService> dataLifecycleServices = internalCluster().getInstances(DataLifecycleService.class);
 
         String dataStreamName = "metrics-foo";
@@ -451,16 +459,7 @@ public class DataLifecycleServiceIT extends ESIntegTestCase {
             Iterable<DataLifecycleService> lifecycleServices = internalCluster().getInstances(DataLifecycleService.class);
 
             for (DataLifecycleService lifecycleService : lifecycleServices) {
-                if (internalCluster().numDataNodes() > 1) {
-                    assertThat(lifecycleService.getErrorStore().getError(previousWriteInddex), nullValue());
-                } else {
-                    if (lifecycleService.getErrorStore().getError(previousWriteInddex) != null) {
-                        assertThat(
-                            lifecycleService.getErrorStore().getError(previousWriteInddex),
-                            containsString("Force merge request only had 1 successful shards out of a total of 2")
-                        );
-                    }
-                }
+                assertThat(lifecycleService.getErrorStore().getError(previousWriteInddex), nullValue());
             }
         });
     }
@@ -470,7 +469,14 @@ public class DataLifecycleServiceIT extends ESIntegTestCase {
         // that its retention execution fails
         DataLifecycle lifecycle = new DataLifecycle();
 
-        putComposableIndexTemplate("id1", null, List.of("metrics-foo*"), null, null, lifecycle);
+        putComposableIndexTemplate(
+            "id1",
+            null,
+            List.of("metrics-foo*"),
+            Settings.builder().put(IndexMetadata.SETTING_AUTO_EXPAND_REPLICAS, "0-1").build(),
+            null,
+            lifecycle
+        );
         Iterable<DataLifecycleService> dataLifecycleServices = internalCluster().getInstances(DataLifecycleService.class);
 
         String dataStreamName = "metrics-foo";
@@ -541,16 +547,7 @@ public class DataLifecycleServiceIT extends ESIntegTestCase {
                 // error stores don't contain anything for the first generation index anymore
                 Iterable<DataLifecycleService> lifecycleServices = internalCluster().getInstances(DataLifecycleService.class);
                 for (DataLifecycleService lifecycleService : lifecycleServices) {
-                    if (internalCluster().numDataNodes() > 1) {
-                        assertThat(lifecycleService.getErrorStore().getError(firstGenerationIndex), nullValue());
-                    } else {
-                        if (lifecycleService.getErrorStore().getError(firstGenerationIndex) != null) {
-                            assertThat(
-                                lifecycleService.getErrorStore().getError(firstGenerationIndex),
-                                containsString("Force merge request only had 1 successful shards out of a total of 2")
-                            );
-                        }
-                    }
+                    assertThat(lifecycleService.getErrorStore().getError(firstGenerationIndex), nullValue());
                 }
             });
         } finally {

--- a/modules/dlm/src/internalClusterTest/java/org/elasticsearch/dlm/DataLifecycleServiceIT.java
+++ b/modules/dlm/src/internalClusterTest/java/org/elasticsearch/dlm/DataLifecycleServiceIT.java
@@ -385,7 +385,10 @@ public class DataLifecycleServiceIT extends ESIntegTestCase {
     public void testErrorRecordingOnRollover() throws Exception {
         // empty lifecycle contains the default rollover
         DataLifecycle lifecycle = new DataLifecycle();
-
+        /*
+         * We set index.auto_expand_replicas to 0-1 so that if we get a single-node cluster it is not yellow. The cluster being yellow
+         * could result in DLM's automatic forcemerge failing, which would result in an unexpected error in the error store.
+         */
         putComposableIndexTemplate(
             "id1",
             null,
@@ -469,6 +472,10 @@ public class DataLifecycleServiceIT extends ESIntegTestCase {
         // that its retention execution fails
         DataLifecycle lifecycle = new DataLifecycle();
 
+        /*
+         * We set index.auto_expand_replicas to 0-1 so that if we get a single-node cluster it is not yellow. The cluster being yellow
+         * could result in DLM's automatic forcemerge failing, which would result in an unexpected error in the error store.
+         */
         putComposableIndexTemplate(
             "id1",
             null,

--- a/modules/dlm/src/internalClusterTest/java/org/elasticsearch/dlm/DataLifecycleServiceIT.java
+++ b/modules/dlm/src/internalClusterTest/java/org/elasticsearch/dlm/DataLifecycleServiceIT.java
@@ -381,7 +381,6 @@ public class DataLifecycleServiceIT extends ESIntegTestCase {
             .count();
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/96070")
     public void testErrorRecordingOnRollover() throws Exception {
         // empty lifecycle contains the default rollover
         DataLifecycle lifecycle = new DataLifecycle();
@@ -452,7 +451,16 @@ public class DataLifecycleServiceIT extends ESIntegTestCase {
             Iterable<DataLifecycleService> lifecycleServices = internalCluster().getInstances(DataLifecycleService.class);
 
             for (DataLifecycleService lifecycleService : lifecycleServices) {
-                assertThat(lifecycleService.getErrorStore().getError(previousWriteInddex), nullValue());
+                if (internalCluster().numDataNodes() > 1) {
+                    assertThat(lifecycleService.getErrorStore().getError(previousWriteInddex), nullValue());
+                } else {
+                    if (lifecycleService.getErrorStore().getError(previousWriteInddex) != null) {
+                        assertThat(
+                            lifecycleService.getErrorStore().getError(previousWriteInddex),
+                            containsString("Force merge request only had 1 successful shards out of a total of 2")
+                        );
+                    }
+                }
             }
         });
     }

